hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/BlockStreamStateManager.java

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.blocks.impl.streaming;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.block.BlockItemSet;
+import com.hedera.hapi.block.PublishStreamRequest;
+import com.hedera.hapi.block.stream.BlockItem;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Manages the state of blocks being streamed to block nodes.
+ * This class is responsible for maintaining the block states and providing methods for adding items to blocks
+ * and creating requests.
+ */
+public class BlockStreamStateManager {
```

Comment:
We already have lots of references to 'block' and 'stream'. 

```suggestion
public class BlockToNodesStateManager {
```

or

```suggestion
public class StreamedBlocksStateManager {
```

Reply:
Also `BlockNodesStateManager` could be a candidate, we can think about it